src/integration/integration_api_scalingengine_test.go

```diff
@@ -47,6 +48,52 @@ var _ = Describe("Integration_Api_ScalingEngine", func() {
 		stopScalingEngine()
 	})
 	Describe("Get scaling histories", func() {
+
+		Context("Cloud Controller api is not available", func() {
+			BeforeEach(func() {
+				fakeCCNOAAUAA.Reset()
+				fakeCCNOAAUAA.AllowUnhandledRequests = true
+				parameters = map[string]string{"start-time": "1111", "end-time": "9999", "order": "desc", "page": "1", "results-per-page": "5"}
+			})
+			It("should error when access public api", func() {
+				By("check public api")
+				checkResponseContentWithParameters(getScalingHistories, pathVariables, parameters, http.StatusBadRequest, map[string]interface{}{}, PUBLIC)
+			})
+		})
+
+		Context("UAA api is not available", func() {
+			BeforeEach(func() {
+				fakeCCNOAAUAA.Reset()
+				fakeCCNOAAUAA.AllowUnhandledRequests = true
+				fakeCCNOAAUAA.RouteToHandler("GET", "/v2/info", ghttp.RespondWithJSONEncoded(http.StatusOK,
+					cf.Endpoints{
+						AuthEndpoint:    fakeCCNOAAUAA.URL(),
+						DopplerEndpoint: strings.Replace(fakeCCNOAAUAA.URL(), "http", "ws", 1),
+					}))
+				parameters = map[string]string{"start-time": "1111", "end-time": "9999", "order": "desc", "page": "1", "results-per-page": "5"}
+			})
+			It("should error when access public api", func() {
+				By("check public api")
```

Comment:
as explained in app.js,  it is incorrect to return 400 bad request when  "cc /uaa unavailable"